### PR TITLE
Bugfix/v1-119 | Blank test result page

### PR DESCRIPTION
**Pull request by dapodev**
# Overview
>
> - Fixed problems with blank page.
> - Fixed issues with skills populating.
> - Fixed type problems.

### CODE REVIEW

File: backend/src/utils/comparator/ObjectId/containsObjectId.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { ObjectId } from 'mongoose';
+
+const containsObjectId = (array: ObjectId[], searchValue: ObjectId): boolean =>
+  !!array.filter((objId) => searchValue.toString() === objId.toString()).length;
```
Comment (SaXaPhonist) [inline backend/src/utils/comparator/ObjectId/containsObjectId.ts:4]: array.some()

File: backend/src/db/providers/skillProvider.ts [hunk 2]
```diff
@@ -1,23 +1,102 @@
...
+
+  if (!updatedUserSkill) {
+    throw new NotFoundError('User skill not found.');
+  }
+
+  return updatedUserSkill;
+};
+
+const populateUserSkills = async (userSkills: IUserSkill[]): Promise<IUserSkill[]> => {
+  const mappedUserSkills = userSkills.map((uskill) => {
```
Comment (SaXaPhonist) [inline backend/src/db/providers/skillProvider.ts:82]: excess

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: backend/src/db/providers/skillProvider.ts [hunk 3]
```diff
@@ -1,23 +1,102 @@
...
+  const skills: IUserSkill[] = await UserSkillModel.find({ user: userId })
+    .select('-_id skill score')
+    .lean();
   return skills;
 };
 
-const addUserSkill = async (userId: string, skillId?: ObjectId) => {
-  await UserSkillModel.create({
-    user: userId,
+const attachSkillToUserProfile = async (userId: string, userSkillId: ObjectId) => {
```
Comment (SaXaPhonist) [inline backend/src/db/providers/skillProvider.ts:17]: move all logic out of provider, follow plz single responsibility priciple

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: backend/src/db/providers/skillProvider.ts [hunk 4]
```diff
@@ -1,23 +1,102 @@
...
+
+const addUserSkill = async (userId: string, skillId?: ObjectId): Promise<IUserSkill> => {
+  const insertedUserSkill: IUserSkill = await UserSkillModel.create({
+    user: new mongoose.Types.ObjectId(userId),
     skill: skillId,
     score: 1,
   });
+
+  if (insertedUserSkill._id) {
+    await attachSkillToUserProfile(userId, insertedUserSkill._id);
```
[the remark this answers is not in the document]
Comment (SaXaPhonist) [reply backend/src/db/providers/skillProvider.ts:61]: call attachSkillToUser from controller not from another provider

File: backend/src/db/providers/skillProvider.ts (hunk 3 above)
Comment (SaXaPhonist) [reply backend/src/db/providers/skillProvider.ts:17]: for example we already have provider that return user info, we need to use it cause it was the point of creation single provider, use them from controller, then move all helpers method into helpers
if attachSkillToUser means update user data create method update and unique provider for that

File: backend/src/db/providers/skillProvider.ts (hunk 2 above)
Comment (SaXaPhonist) [reply backend/src/db/providers/skillProvider.ts:82]: try to mapped before come to provider

File: backend/src/db/providers/skillProvider.ts (hunk 3 above)
Comment (k-krasilnikova) [reply backend/src/db/providers/skillProvider.ts:17]: Agree with @SaXaPhonist

File: backend/src/utils/normaliser/skills.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { IUserSkill } from 'interfaces/Ientities/IUserSkill';
+
+const extractCommonUserSkillInfo = (skills: IUserSkill[]) =>
+  skills.map((uSkill) => {
+    return { score: uSkill.score, skill: uSkill.skill };
```
Comment (SaXaPhonist) [inline backend/src/utils/normaliser/skills.ts:5]: also you can  `.map(uSkill) => ( {score: ... , skill: ...} )`

**Review by SaXaPhonist — DISMISSED** [review]

File: backend/src/utils/normaliser/skills.ts (hunk 5 above)
Comment (dapodev) [reply backend/src/utils/normaliser/skills.ts]: I HAVEN'T EVER HEARD ABOUT THAT (ONE LINE OBJECT RETURN). THANKS 😆 😆 😆

**Pull request merged**